### PR TITLE
*: add global variable tidb_slow_log_masking to control masking slow log query

### LINKED ISSUE #17463: Flag to not print arguments in slowlog
We would like a flag that disables the printing of argument values in the slowlog. Only the query should be printed.

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs <crazycs520@gmail.com>

### What problem does this PR solve?

Issue Number: close  #17463

This PR adds global variable `tidb_slow_log_masking` to control masking slow log query. 

Such as below:

1. Enable the  `tidb_slow_log_masking` first.

```sql
 set @@global.tidb_slow_log_masking=1;
```

2. Then, when TiDB record the slow log, it will masking the user data of SQL, 
just like SQL normalize, such as below:

```sql
# Time: 2020-06-03T18:28:39.886171+08:00
...
...
...
select count ( ? ) , sum ( query_time ) , digest from information_schema . cluster_slow_query where time >= ? and time < ? group by digest order by sum ( query_time ) desc;
```

The original SQL is:

```sql
select count(*), sum(query_time), digest from information_schema.CLUSTER_SLOW_QUERY where time >= '2020-03-05 20:55:00' and time < '2020-06-05 20:57:00' group by digest order by sum(query_time) desc;
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:

### Check List 

Tests 

- Unit test
- Manual test

Side effects

- No

### Release note 

- 

- add global variable tidb_slow_log_masking to control masking slow log query.

### CODE REVIEW

**Comment by bb7133** [comment]
Data Race: https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/38689/pipeline

File: sessionctx/variable/tidb_vars.go [hunk 1]
```diff
@@ -399,6 +399,9 @@ const (
 
 	// TiDBEnableClusteredIndex indicates if clustered index feature is enabled.
 	TiDBEnableClusteredIndex = "tidb_enable_clustered_index"
+
+	// TiDBSlowLogMasking indicates that whether masking the query data when log slow query.
```
Comment (breeswish) [inline sessionctx/variable/tidb_vars.go:403]: Does it only effective for parameterized queries?

Comment (crazycs520) [reply sessionctx/variable/tidb_vars.go:403]: What's your mean of `parameterized queries`?

Actually the masking SQL is also used to generate the SQL digest.

Comment (breeswish) [reply sessionctx/variable/tidb_vars.go:403]: Oh, I mean prepared statements. For non prepared statements, will they have effect?

**Comment by breeswish** [comment]
What will happen to the parameters appears in the plan? This can also be a source of leaking data.

File: sessionctx/variable/tidb_vars.go (hunk 1 above)
Comment (crazycs520) [reply sessionctx/variable/tidb_vars.go:403]: yes, It also works on non-prepared statement.

**Pull request merged**